File: Python/Day05_Patterns1.py
"""
Day 05: Hollow Rectangle Pattern
Printing a rectangle with only border stars.
"""

n = int(input("Enter the number of rows:"))
m = int(input("Enter the number of columns:"))

for i in range(1, n + 1):
    for j in range(1, m + 1):
        # Print star for borders, space for inside
        if i == 1 or i == n or j == 1 or j == m:
            print("*", end="")
        else:
            print(" ", end="")
    print() # Newline after each row

File: Java/Day05_Patterns1.java
/**
 * Day 05: Hollow Rectangle Pattern
 * Printing a rectangle with only border stars.
 */

import java.util.*;

public class Day05_Patterns1 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the number of rows:");
        int n = sc.nextInt();
        System.out.println("Enter the number of columns:");
        int m = sc.nextInt();
  
        for (int i = 1; i <= n; i++) {
            for (int j = 1; j <= m; j++) {
                if (i == 1 || i == n || j == 1 || j == m) {
                    System.out.print("*");
                }else{
                    System.out.print(" ");
                }
            }
            System.out.println();

        }

    }

}
